src/Datastore/DatastoreClient.php

```diff
@@ -0,0 +1,796 @@
+            : null;
+    }
+
+    /**
+     * Get multiple entities
+     *
+     * Example:
+     * ```
+     * $keys = [
+     *     $datastore->key('Person', 'Bob'),
+     *     $datastore->key('Person', 'John')
+     * ];
+     *
+     * $entities = $datastore->lookup($keys);
+     *
+     * foreach ($entities['found'] as $entity) {
+     *     echo $entity['firstName'];
+     * }
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @param Key[] $key The identifiers to look up.
+     * @param array $options {
+     *     Configuration Options
+     *
+     *     @type string $readConsistency See
+     *           [ReadConsistency](https://cloud.google.com/datastore/reference/rest/v1beta3/ReadOptions#ReadConsistency).
+     *     @type string $className The name of the class to return results as.
+     *           Must be a subclass of {@see Google\Cloud\Datastore\Entity}.
+     *           If not set, {@see Google\Cloud\Datastore\Entity} will be used.
+     * }
+     * @return array
+     * @codingStandardsIgnoreEnd
+     */
+    public function lookupBatch(array $keys, array $options = [])
+    {
+        return $this->operation->lookup($keys, $options);
+    }
+
+    /**
+     * Create a Query
+     *
+     * The Query class can be used as a builder, or it can accept a query
+     * representation at instantiation.
+     *
+     * Example:
+     * ```
+     * $query = $datastore->query();
+     * ```
+     *
+     * @param array $options {
+     *     Query Options
+     *
+     *     @type array $query [Query](https://cloud.google.com/datastore/reference/rest/v1beta3/projects/runQuery#query)
+     * }
+     * @return Query
+     */
+    public function query(array $options = [])
+    {
+        return new Query($this->projectId, $options);
+    }
+
+    /**
+     * Create a GqlQuery
+     *
+     * Example:
+     * ```
+     * $query = $datastore->gqlQuery('SELECT * FROM people');
+     * ```
+     *
+     * ```
+     * // Literals must be provided as bound parameters by default:
+     * $query = $datastore->gqlQuery('SELECT * FROM people WHERE firstName = @firstName', [
+     *     'bindings' => [
+     *         'firstName' => 'Bob'
+     *     ]
+     * ]);
+     * ```
+     *
+     * ```
+     * // Positional binding is also supported
+     * $query = $datastore->gqlQuery('SELECT * FROM Companies WHERE companyName = @1 LIMIT 1', [
+     *     'bindings' => [
+     *         'Google'
+     *     ]
+     * ]);
+     * ```
+     *
+     * ```
+     * // While not recommended, you can use literals in your query string:
+     * $query = $datastore->gqlQuery("SELECT * FROM people WHERE firstName = 'Bob'", [
+     *     'allowLiterals' => true
+     * ]);
+     * ```
+     *
+     * @param string $query The [GQL Query](https://cloud.google.com/datastore/docs/apis/gql/gql_reference) string
+     * @param array $options See {@see Google\Cloud\Datastore\Query\GqlQuery::__construct()} for supported options.
```

Comment:
I think it is worth explicitly documenting the options available here instead of linking out to the constructor. Also, since we can potentially know the namespace up front due to it being provided in `DatastoreClient::construct()` we should default that (the same may apply to `DatastoreClient::query()` as well?).